app/utils/computed-helpers.js

```diff
@@ -47,13 +47,31 @@ export const computedSegmentedLink = function(property) {
 export const computedDateTimeSplit = function(property, segmentFormat, endProperty) {
   return computed(property, {
     get() {
-      return moment(this.get(property)).format(getFormat(segmentFormat));
+      let timezone = '';
+      if (this.constructor.modelName === 'event') {
+        timezone = this.timezone;
+      } else {
+        timezone = this.event.get('timezone');
+      }
+      return moment(this.get(property)).tz(timezone).format(getFormat(segmentFormat));
     },
     set(key, value) {
-      const newDate = moment(value, getFormat(segmentFormat));
+      let timezone = '';
+      if (this.constructor.modelName === 'event') {
+        timezone = this.timezone;
+      } else {
+        timezone = this.event.get('timezone');
+      }
+      let newDate = moment(value, getFormat(segmentFormat)).tz(timezone, true);
       let oldDate = newDate;
       if (this.get(property)) {
-        oldDate = moment(this.get(property), segmentFormat === 'date' ? FORM_DATE_FORMAT : FORM_TIME_FORMAT);
+        timezone = '';
+        if (this.constructor.modelName === 'event') {
+          timezone = this.timezone;
+        } else {
+          timezone = this.event.get('timezone');
+        }
```

Comment:
I don't even understand why this block of code is repeated here in this if statement.